File: backend/app/models/candidate_evaluation.py
from sqlalchemy import Column, String, DateTime, ForeignKey, JSON, Text
from datetime import datetime
import uuid
from ..database import db

class CandidateEvaluation(db.Model):
    __tablename__ = 'candidate_evaluation'

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    candidate_profile_id = Column(String(36), ForeignKey('candidate_profile.id'), nullable=False, unique=True)

    # Post-assessment data
    skills_with_level = Column(JSON, nullable=True)  # e.g., [{"skill": "Excel", "level": "Intermediate"}]
    suitable_training = Column(String(200), nullable=True)
    comments = Column(Text, nullable=True)
    remarks = Column(Text, nullable=True)

    status = Column(String(50), nullable=False, default='Under Review')  # Options: "Selected", "Rejected", "Under Review"
    # assigned_job_role = Column(String(50), nullable=True)

    created_by = Column(String(36), ForeignKey('users.id'), nullable=False)
    updated_by = Column(String(36), ForeignKey('users.id'), nullable=True)
    
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

File: frontend/src/models/candidate_evaluation.ts
export interface CandidateEvaluation {
  id: string;
  candidate_profile_id: string;
  skills_with_level?: string;
  suitable_training?: string;
  comments?: string;
  remarks?: string;
  status: string;
  created_by?: UserInfo | null;
  updated_by?: UserInfo | null;
  created_at?: string;
  updated_at?: string;
}

export interface CreateCandidateEvaluationRequest {
  candidate_profile_id: string;
  skills_with_level?: string;
  suitable_training?: string;
  comments?: string;
  remarks?: string;
  status: string;
}

export interface UpdateCandidateEvaluationRequest {
  skills_with_level?: string;
  suitable_training?: string;
  comments?: string;
  remarks?: string;
  status?: string;
}

export interface CandidateEvaluationResponse {
  status: "success" | "error";
  message?: string;
  data?: CandidateEvaluation;
  updated_by?: UserInfo;
  created_by?: UserInfo;
}

export interface CandidateEvaluationListResponse {
  status: "success" | "error";
  data: CandidateEvaluation[];
}

export interface UserInfo {
  id: string;
  name: string;
  email?: string;
  role?: string;
}
